### ISSUE TITLE
Url in tasks.org show as weird day header

### ISSUE THREAD

**Issue by damionx7** [andstatus/todoagenda#49]
Hey there, I have saved many tasks with title as a url. Such as this one. ![Screenshot_20200912-153027.jpg](https://user-images.githubusercontent.com/6530843/92993084-e8100c80-f50c-11ea-87a9-48e66832d824.jpg)

And these show up as tasks in todo agenda widget as 

![Screenshot_20200912-153058.jpg](https://user-images.githubusercontent.com/6530843/92993105-01b15400-f50d-11ea-860c-2735381593f8.jpg)

How can i solve this ?

**Comment by yvolk** [comment]
@damionx7 In order to help me reproduce the problem could you attach here or send me by email a snapshot of these events:
Settings -> Feedback and Backup -> Share events and settings for debugging

**Comment by damionx7** [comment]
The app crashes when I click Share Events and settings 

![Screenshot_20200912-202701.jpg](https://user-images.githubusercontent.com/6530843/92998252-6503ac00-f536-11ea-8a49-e5fbafbbcb38.jpg)

**Comment by yvolk** [comment]
This means that problem is deeper, and this explains the weird day header.
Could you try to change filters, maybe create separate Task list for testing in order to figure out, what exactly event causes such a crash?
I tried to add a task with URL as a title just like you did - no problem...

**Comment by damionx7** [comment]
@yvolk I have sent you an email with video of how removing the url solves the issue. Kindly check the first task in video.

**Comment by yvolk** [comment]
@damionx7 Looking at your video I see "??? DAY_HEADER" both before and after removal of URL, so looks like the issue is not fixed by this change...

**Comment by damionx7** [comment]
I mean the task is removed from the ??? Day header.

**Comment by yvolk** [comment]
@damionx7 Your video shows that the task is not present in its previous place. But as the list is too long, we don't see, where it is now.
In order to localize a problem, please do the following:

1. Create in Tasks app new List and call it "Widget testing".
2. Create new "ToDo Agenda" widget and in its settings "Calendars and Task Lists" select only the "Widget testing" list. Don't touch any other settings.
3. Make sure the widget works and doesn't show any events and other unexpected data.
4. Add ONE task to the "Widget testing" list with title: "Task in Widget testing".
5. Make sure the widget works and shows the new task normally. Make a screenshot and attach here.
6. Attach here or send me by email a snapshot of the event and settings:
Settings -> Feedback and Backup -> Share events and settings for debugging.
7. Add the second task to the "Widget testing" list with URL title.
8. Make sure the widget works and shows two tasks normally. Make a screenshot and attach here.
9. Attach here or send me by email a snapshot of events and settings:
Settings -> Feedback and Backup -> Share events and settings for debugging.

**Comment by damionx7** [comment]
## Scenario 1: only one task

Screenshot:

![Screenshot_20200913-112751.jpg](https://user-images.githubusercontent.com/6530843/93011534-958a2b00-f5b4-11ea-9b24-61afbc58f675.jpg)

Logs:
```
{
  "deviceInfo": {
    "versionCode": 29,
    "versionRelease": "10",
    "versionCodename": "REL",
    "buildManufacturer": "OnePlus",
    "buildBrand": "OnePlus",
    "buildModel": "GM1911"
  },
  "applicationInfo": {
    "versionName": "4.2.3",
    "versionCode": 686
  },
  "settings": {
    "widgetId": 14,
    "widgetHeaderDateFormat": "defaultWeekday:",
    "widgetInstanceName": "Todo Agenda 14",
    "activeSources": [
      {
        "providerType": 4,
        "id": 15,
        "title": "Widget Testing",
        "summary": "",
        "color": 0,
        "isAvailable": true
      }
    ],
    "eventRange": 30,
    "eventsEnded": "NONE",
    "fillAllDay": true,
    "hideBasedOnKeywords": "",
    "widgetHeaderBackgroundColor": 0,
    "pastEventsBackgroundColor": -1082623956,
    "todaysEventsBackgroundColor": -620756993,
    "backgroundColor": -2147483648,
    "showDaysWithoutEvents": false,
    "showDayHeaders": true,
    "dayHeaderDateFormat": "defaultWeekday:",
    "horizontalLineBelowDayHeader": false,
    "showPastEventsUnderOneHeader": false,
    "showPastEventsWithDefaultColor": false,
    "showEventIcon": true,
    "entryDateFormat": "days:bbbb",
    "showEndTime": true,
    "showLocation": true,
    "dateFormat": "auto",
    "lockedTimeZoneId": "",
    "snapshotMode": "live_data",
    "refreshPeriodMinutes": 10,
    "eventEntryLayout": "DEFAULT",
    "multiline_title": false,
    "multiline_details": false,
    "showOnlyClosestInstanceOfRecurringEvent": false,
    "hideDuplicates": false,
    "taskScheduling": "date_due",
    "taskWithoutDates": "end_of_list",
    "filterMode": "normal",
    "indicateAlerts": true,
    "indicateRecurring": false,
    "compactLayout": false,
    "widgetHeaderLayout": "ONE_ROW",
    "headerTheme": "DARK",
    "dayHeaderThemePast": "DARK",
    "entryThemePast": "BLACK",
    "dayHeaderTheme": "DARK",
    "entryTheme": "DARK",
    "dayHeaderThemeFuture": "LIGHT",
    "entryThemeFuture": "DARK",
    "textSizeScale": "1.0",
    "dayHeaderAlignment": "RIGHT"
  },
  "resultsVersion": 3,
  "results": [
    {
      "providerType": 4,
      "executedAt": 1599976688638,
      "timeZoneId": "Asia\/Kolkata",
      "millisOffsetUtcToLocal": 19800000,
      "standardMillisOffsetUtcToLocal": 19800000,
      "uri": "content:\/\/org.tasks\/todoagenda",
      "projection": [
        "gtl_account",
        "cd_object",
        "parent",
        "cd_order",
        "notes",
        "dueDate",
        "gt_last_sync",
        "cd_vtodo",
        "cd_last_sync",
        "cdl_id",
        "cdl_uuid",
        "gt_parent",
        "snoozeTime",
        "gt_task",
        "cdl_url",
        "cd_id",
        "modified",
        "cdl_color",
        "cd_deleted",
        "cdl_name",
        "estimatedSeconds",
        "cd_etag",
        "gt_order",
        "created",
        "completed",
        "repeatUntil",
        "recurrence",
        "parent_uuid",
        "notificationFlags",
        "gtl_color",
        "cdl_icon",
        "cd_calendar",
        "gtl_icon",
        "_id",
        "lastNotified",
        "cd_task",
        "notifications",
        "cdl_order",
        "timerStart",
        "importance",
        "collapsed",
        "cd_remote_id",
        "title",
        "gt_id",
        "gt_remote_id",
        "gt_deleted",
        "gtl_remote_id",
        "cdl_ctag",
        "calendarUri",
        "gtl_title",
        "elapsedSeconds",
        "gt_moved",
        "gt_remote_order",
        "cd_remote_parent",
        "gtl_id",
        "gtl_remote_order",
        "hideUntil",
        "remoteId",
        "deleted",
        "gt_remote_parent",
        "gtl_last_sync",
        "cdl_account",
        "gt_list_id"
      ],
      "selection": "completed = 0 AND hideUntil <= 1602568688638 AND cdl_id IN ( 15 )",
      "selectionArgs": [],
      "sortOrder": "",
      "rows": [
        {
          "cd_object": {
            "type": 3,
            "value": "1502853656344784351.ics"
          },
          "parent": {
            "type": 1,
            "value": 0
          },
          "dueDate": {
            "type": 1,
            "value": 0
          },
          "cd_last_sync": {
            "type": 1,
            "value": 0
          },
          "cdl_id": {
            "type": 1,
            "value": 15
          },
          "cdl_uuid": {
            "type": 3,
            "value": "3501830923649624372"
          },
          "snoozeTime": {
            "type": 1,
            "value": 0
          },
          "cd_id": {
            "type": 1,
            "value": 3765
          },
          "modified": {
            "type": 1,
            "value": 1599976644323
          },
          "cdl_color": {
            "type": 1,
            "value": 0
          },
          "cd_deleted": {
            "type": 1,
            "value": 0
          },
          "cdl_name": {
            "type": 3,
            "value": "Widget Testing"
          },
          "estimatedSeconds": {
            "type": 1,
            "value": 0
          },
          "created": {
            "type": 1,
            "value": 1599976638068
          },
          "completed": {
            "type": 1,
            "value": 0
          },
          "repeatUntil": {
            "type": 1,
            "value": 0
          },
          "notificationFlags": {
            "type": 1,
            "value": 6
          },
          "cdl_icon": {
            "type": 1,
            "value": -1
          },
          "cd_calendar": {
            "type": 3,
            "value": "3501830923649624372"
          },
          "_id": {
            "type": 1,
            "value": 3357
          },
          "lastNotified": {
            "type": 1,
            "value": 0
          },
          "cd_task": {
            "type": 1,
            "value": 3357
          },
          "notifications": {
            "type": 1,
            "value": 0
          },
          "cdl_order": {
            "type": 1,
            "value": -1
          },
          "timerStart": {
            "type": 1,
            "value": 0
          },
          "importance": {
            "type": 1,
            "value": 2
          },
          "collapsed": {
            "type": 1,
            "value": 0
          },
          "cd_remote_id": {
            "type": 3,
            "value": "1502853656344784351"
          },
          "title": {
            "type": 3,
            "value": "Task in widget testing"
          },
          "elapsedSeconds": {
            "type": 1,
            "value": 0
          },
          "hideUntil": {
            "type": 1,
            "value": -86400000
          },
          "remoteId": {
            "type": 3,
            "value": "1910907419188615064"
          },
          "deleted": {
            "type": 1,
            "value": 0
          },
          "cdl_account": {
            "type": 3,
            "value": "local"
          }
        }
      ]
    }
  ]
}

```

## Scenario 2: with url task

Screenshot:
![Screenshot_20200913-112914.jpg](https://user-images.githubusercontent.com/6530843/93011548-b94d7100-f5b4-11ea-9dc4-22461503db28.jpg)

Logs:

```
{
  "deviceInfo": {
    "versionCode": 29,
    "versionRelease": "10",
    "versionCodename": "REL",
    "buildManufacturer": "OnePlus",
    "buildBrand": "OnePlus",
    "buildModel": "GM1911"
  },
  "applicationInfo": {
    "versionName": "4.2.3",
    "versionCode": 686
  },
  "settings": {
    "widgetId": 14,
    "widgetHeaderDateFormat": "defaultWeekday:",
    "widgetInstanceName": "Todo Agenda 14",
    "activeSources": [
      {
        "providerType": 4,
        "id": 15,
        "title": "Widget Testing",
        "summary": "",
        "color": 0,
        "isAvailable": true
      }
    ],
    "eventRange": 30,
    "eventsEnded": "NONE",
    "fillAllDay": true,
    "hideBasedOnKeywords": "",
    "widgetHeaderBackgroundColor": 0,
    "pastEventsBackgroundColor": -1082623956,
    "todaysEventsBackgroundColor": -620756993,
    "backgroundColor": -2147483648,
    "showDaysWithoutEvents": false,
    "showDayHeaders": true,
    "dayHeaderDateFormat": "defaultWeekday:",
    "horizontalLineBelowDayHeader": false,
    "showPastEventsUnderOneHeader": false,
    "showPastEventsWithDefaultColor": false,
    "showEventIcon": true,
    "entryDateFormat": "days:bbbb",
    "showEndTime": true,
    "showLocation": true,
    "dateFormat": "auto",
    "lockedTimeZoneId": "",
    "snapshotMode": "live_data",
    "refreshPeriodMinutes": 10,
    "eventEntryLayout": "DEFAULT",
    "multiline_title": false,
    "multiline_details": false,
    "showOnlyClosestInstanceOfRecurringEvent": false,
    "hideDuplicates": false,
    "taskScheduling": "date_due",
    "taskWithoutDates": "end_of_list",
    "filterMode": "normal",
    "indicateAlerts": true,
    "indicateRecurring": false,
    "compactLayout": false,
    "widgetHeaderLayout": "ONE_ROW",
    "headerTheme": "DARK",
    "dayHeaderThemePast": "DARK",
    "entryThemePast": "BLACK",
    "dayHeaderTheme": "DARK",
    "entryTheme": "DARK",
    "dayHeaderThemeFuture": "LIGHT",
    "entryThemeFuture": "DARK",
    "textSizeScale": "1.0",
    "dayHeaderAlignment": "RIGHT"
  },
  "resultsVersion": 3,
  "results": [
    {
      "providerType": 4,
      "executedAt": 1599976763101,
      "timeZoneId": "Asia\/Kolkata",
      "millisOffsetUtcToLocal": 19800000,
      "standardMillisOffsetUtcToLocal": 19800000,
      "uri": "content:\/\/org.tasks\/todoagenda",
      "projection": [
        "gtl_account",
        "cd_object",
        "parent",
        "cd_order",
        "notes",
        "dueDate",
        "gt_last_sync",
        "cd_vtodo",
        "cd_last_sync",
        "cdl_id",
        "cdl_uuid",
        "gt_parent",
        "snoozeTime",
        "gt_task",
        "cdl_url",
        "cd_id",
        "modified",
        "cdl_color",
        "cd_deleted",
        "cdl_name",
        "estimatedSeconds",
        "cd_etag",
        "gt_order",
        "created",
        "completed",
        "repeatUntil",
        "recurrence",
        "parent_uuid",
        "notificationFlags",
        "gtl_color",
        "cdl_icon",
        "cd_calendar",
        "gtl_icon",
        "_id",
        "lastNotified",
        "cd_task",
        "notifications",
        "cdl_order",
        "timerStart",
        "importance",
        "collapsed",
        "cd_remote_id",
        "title",
        "gt_id",
        "gt_remote_id",
        "gt_deleted",
        "gtl_remote_id",
        "cdl_ctag",
        "calendarUri",
        "gtl_title",
        "elapsedSeconds",
        "gt_moved",
        "gt_remote_order",
        "cd_remote_parent",
        "gtl_id",
        "gtl_remote_order",
        "hideUntil",
        "remoteId",
        "deleted",
        "gt_remote_parent",
        "gtl_last_sync",
        "cdl_account",
        "gt_list_id"
      ],
      "selection": "completed = 0 AND hideUntil <= 1602568763101 AND cdl_id IN ( 15 )",
      "selectionArgs": [],
      "sortOrder": "",
      "rows": [
        {
          "cd_object": {
            "type": 3,
            "value": "1502853656344784351.ics"
          },
          "parent": {
            "type": 1,
            "value": 0
          },
          "dueDate": {
            "type": 1,
            "value": 0
          },
          "cd_last_sync": {
            "type": 1,
            "value": 0
          },
          "cdl_id": {
            "type": 1,
            "value": 15
          },
          "cdl_uuid": {
            "type": 3,
            "value": "3501830923649624372"
          },
          "snoozeTime": {
            "type": 1,
            "value": 0
          },
          "cd_id": {
            "type": 1,
            "value": 3765
          },
          "modified": {
            "type": 1,
            "value": 1599976644323
          },
          "cdl_color": {
            "type": 1,
            "value": 0
          },
          "cd_deleted": {
            "type": 1,
            "value": 0
          },
          "cdl_name": {
            "type": 3,
            "value": "Widget Testing"
          },
          "estimatedSeconds": {
            "type": 1,
            "value": 0
          },
          "created": {
            "type": 1,
            "value": 1599976638068
          },
          "completed": {
            "type": 1,
            "value": 0
          },
          "repeatUntil": {
            "type": 1,
            "value": 0
          },
          "notificationFlags": {
            "type": 1,
            "value": 6
          },
          "cdl_icon": {
            "type": 1,
            "value": -1
          },
          "cd_calendar": {
            "type": 3,
            "value": "3501830923649624372"
          },
          "_id": {
            "type": 1,
            "value": 3357
          },
          "lastNotified": {
            "type": 1,
            "value": 0
          },
          "cd_task": {
            "type": 1,
            "value": 3357
          },
          "notifications": {
            "type": 1,
            "value": 0
          },
          "cdl_order": {
            "type": 1,
            "value": -1
          },
          "timerStart": {
            "type": 1,
            "value": 0
          },
          "importance": {
            "type": 1,
            "value": 2
          },
          "collapsed": {
            "type": 1,
            "value": 0
          },
          "cd_remote_id": {
            "type": 3,
            "value": "1502853656344784351"
          },
          "title": {
            "type": 3,
            "value": "Task in widget testing"
          },
          "elapsedSeconds": {
            "type": 1,
            "value": 0
          },
          "hideUntil": {
            "type": 1,
            "value": -86400000
          },
          "remoteId": {
            "type": 3,
            "value": "1910907419188615064"
          },
          "deleted": {
            "type": 1,
            "value": 0
          },
          "cdl_account": {
            "type": 3,
            "value": "local"
          }
        },
        {
          "cd_object": {
            "type": 3,
            "value": "4262110760745279890.ics"
          },
          "parent": {
            "type": 1,
            "value": 0
          },
          "cd_order": {
            "type": 1,
            "value": 621669437
          },
          "dueDate": {
            "type": 1,
            "value": 0
          },
          "cd_last_sync": {
            "type": 1,
            "value": 0
          },
          "cdl_id": {
            "type": 1,
            "value": 15
          },
          "cdl_uuid": {
            "type": 3,
            "value": "3501830923649624372"
          },
          "snoozeTime": {
            "type": 1,
            "value": 0
          },
          "cd_id": {
            "type": 1,
            "value": 3766
          },
          "modified": {
            "type": 1,
            "value": 1599976737732
          },
          "cdl_color": {
            "type": 1,
            "value": 0
          },
          "cd_deleted": {
            "type": 1,
            "value": 0
          },
          "cdl_name": {
            "type": 3,
            "value": "Widget Testing"
          },
          "estimatedSeconds": {
            "type": 1,
            "value": 0
          },
          "created": {
            "type": 1,
            "value": 1599976714039
          },
          "completed": {
            "type": 1,
            "value": 0
          },
          "repeatUntil": {
            "type": 1,
            "value": 0
          },
          "notificationFlags": {
            "type": 1,
            "value": 6
          },
          "cdl_icon": {
            "type": 1,
            "value": -1
          },
          "cd_calendar": {
            "type": 3,
            "value": "3501830923649624372"
          },
          "_id": {
            "type": 1,
            "value": 3358
          },
          "lastNotified": {
            "type": 1,
            "value": 0
          },
          "cd_task": {
            "type": 1,
            "value": 3358
          },
          "notifications": {
            "type": 1,
            "value": 0
          },
          "cdl_order": {
            "type": 1,
            "value": -1
          },
          "timerStart": {
            "type": 1,
            "value": 0
          },
          "importance": {
            "type": 1,
            "value": 2
          },
          "collapsed": {
            "type": 1,
            "value": 0
          },
          "cd_remote_id": {
            "type": 3,
            "value": "4262110760745279890"
          },
          "title": {
            "type": 3,
            "value": "https:\/\/google.com"
          },
          "elapsedSeconds": {
            "type": 1,
            "value": 0
          },
          "hideUntil": {
            "type": 1,
            "value": -86400000
          },
          "remoteId": {
            "type": 3,
            "value": "877583453880762204"
          },
          "deleted": {
            "type": 1,
            "value": 0
          },
          "cdl_account": {
            "type": 3,
            "value": "local"
          }
        }
      ]
    }
  ]
}
```

**Comment by yvolk** [comment]
Hooray! Thank you very much.
I do see the same weird header now. Will figure out a problem and fix it...

**Comment by yvolk** [comment]
@damionx7 @abaker I fixed the weird "??? DAY_HEADER" day header for events from Astrid Clone Tasks app.
This was caused by unexpected startDate=1969-12-31T05:30:00.000+05:30
(negative milliseconds value of hideUntil: -86400000 when "timeZoneId": "Asia\/Kolkata")

This looks like a bug in Astrid Clone Tasks app, but I defensively fixed this problem at our widget's side anyway by ignoring non-positive values of milliseconds.

**Comment by damionx7** [comment]
Thanks . Acutally I have set the default Astrid Tasks settings to day before due. And when no date is set for new tasks, it might save as negative value.

**Comment by yvolk** [comment]
Released to Beta channel, see #3

**Comment by damionx7** [comment]
This new beta apk solves the issue.
